lib/std/os.zig

```diff
@@ -2443,7 +2445,21 @@ pub const SocketError = error{
     SocketTypeNotSupported,
 } || UnexpectedError;
 
-pub fn socket(domain: u32, socket_type: u32, protocol: u32) SocketError!fd_t {
+pub fn socket(domain: u32, socket_type: u32, protocol: u32) SocketError!socket_t {
+    if (builtin.os.tag == .windows) {
+        // NOTE: cannot remove SOCK_NONBLOCK and SOCK_CLOEXEC from socket_type because
+        //       windows does not define this flags yet
```

Comment:
Use `WSASocketW` instead of `socket` and `SOCK_CLOEXEC` probably corresponds to `WSA_FLAG_NO_HANDLE_INHERIT`?

Reply:
done